crates/orchestrator/src/jobs/mod.rs

```diff
@@ -344,10 +368,17 @@ pub async fn handle_job_failure(id: Uuid, config: Arc<Config>) -> Result<(), Job
     }
 
     metadata.insert("last_job_status".to_string(), job.status.to_string());
-    job.metadata = metadata;
-    job.status = JobStatus::Failed;
-
-    config.database().update_job(&job).await.map_err(|e| JobError::Other(OtherError(e)))?;
+    config
+        .database()
+        .update_job(&job, JobItemUpdates {
```

Comment:
builder pattern?

Reply:
[Here](https://github.com/madara-alliance/madara-orchestrator/pull/134/commits/962ce2151296389dcdd85bcc5ccc1535c1e39efa) you go !